windows/RNGestureHandler/RNGestureHandler/RNGestureHandler.cpp

```diff
@@ -0,0 +1,381 @@
+    UIElement reference = sender.as<winrt::Windows::UI::Xaml::UIElement>();
+
+    auto points = e.GetIntermediatePoints(reference);
+
+    recognizer.ProcessMoveEvents(points);
+}
+
+void RNGestureHandler::onPointerReleased(winrt::Windows::Foundation::IInspectable const& sender,
+    PointerRoutedEventArgs const& e)
+{
+    DEBUG("OnPointerReleased\n");
+
+    // Feed the current point into the gesture recognizer as an up event
+    UIElement reference = sender.as<winrt::Windows::UI::Xaml::UIElement>();
+
+    recognizer.ProcessUpEvent(e.GetCurrentPoint(reference));
+
+    // Release the pointer
+    auto element = reference;
+    element.ReleasePointerCapture(e.Pointer());
+}
+
+void RNGestureHandler::onPointerCanceled(winrt::Windows::Foundation::IInspectable const& sender,
+    PointerRoutedEventArgs const& e)
+{
+    DEBUG("OnPointerCanceled\n");
+
+    recognizer.CompleteGesture();
+
+    UIElement element = sender.as<winrt::Windows::UI::Xaml::UIElement>();
+    element.ReleasePointerCapture(e.Pointer());
+
+    sendEventsInState(RNGestureHandlerState::Cancelled, 0);
+    sendEventsInState(RNGestureHandlerState::Undetermined, 0);
+}
+
+void RNGestureHandler::onManipulationStarted(winrt::Windows::Foundation::IInspectable const&,
+    ManipulationStartedEventArgs const&)
+{
+    DEBUG("onManipulationStarted\n");
+}
+
+void RNGestureHandler::onManipulationUpdated(winrt::Windows::Foundation::IInspectable const&,
+    ManipulationUpdatedEventArgs const&)
+{
+    DEBUG("onManipulationUpdated\n");
+}
+
+void RNGestureHandler::onManipulationCompleted(winrt::Windows::Foundation::IInspectable const&,
+    ManipulationCompletedEventArgs const&)
+{
+    DEBUG("onManipulationCompleted\n");
+}
+
+void RNGestureHandler::onCrossSliding(winrt::Windows::Foundation::IInspectable const&,
+    CrossSlidingEventArgs const&)
+{
+    DEBUG("OnCrossSliding\n");
+}
+
+void RNGestureHandler::onDragging(winrt::Windows::Foundation::IInspectable const&,
+    DraggingEventArgs const&)
+{
+    DEBUG("OnDragging\n");
+}
+
+void RNGestureHandler::onHolding(winrt::Windows::Foundation::IInspectable const&,
+    HoldingEventArgs const&)
+{
+    DEBUG("OnHolding\n");
+}
+
+void RNGestureHandler::onTapped(winrt::Windows::Foundation::IInspectable const&,
+    TappedEventArgs const&)
+{
+    DEBUG("OnTapped\n");
+}
+
+void RNGestureHandler::updateConfig(JSValueObject config)
+{
+
+}
+
+void RNGestureHandler::sendEventsInState(RNGestureHandlerState state,
+    RNGestureHandlerEventExtraData* extraData)
+{
+    if (state == lastState)
+        return;
+
+    RNGestureHandlerEvent event{};
+    event.viewTag = viewTag;
+    event.handlerTag = handlerTag;
+    event.prevState = lastState;
+    event.extraData = extraData;
+
+    if (state == RNGestureHandlerState::Active)
+    {
+        // Generate a unique coalescing-key each time the gesture-handler becomes active. 
+        static uint16_t nextEventCoalescingKey = 0;
```

Comment:
Why it's `uint16_t` (i.e. not wider)?

Reply:
I took it from [iOS implementation](https://github.com/software-mansion/react-native-gesture-handler/blob/e6f11644de4becc72570561fc35f445fa535d684/ios/RNGestureHandler.m#L175), but can make it wider.